src/vm/runners/cairo_runner.rs

```diff
@@ -776,16 +776,25 @@ impl CairoRunner {
         Ok(())
     }
 
-    pub fn relocate(&mut self, vm: &mut VirtualMachine) -> Result<(), TraceError> {
+    pub fn relocate(
+        &mut self,
+        vm: &mut VirtualMachine,
+        relocate_mem: bool,
```

Comment:
It's probably better to just check the condition before calling this.

Reply:
It also relocates the trace, thats why

Reply:
Would it be too bad to split those?

Reply:
We can do it, but it would mean replacing one function call with 3 (compute_effective_sizes - relocate_trace - relocate-memory), and making the user responsible for calling compute_effective_sizes (which will cause the other two to fail if not called)